File: iosApp/iosApp/Model/QuestionChoice.swift
//
//  QuestionChoice.swift
//  CACDGAME
//
//  Created by François Dabonot on 03/07/2023.
//  Copyright © 2023 CACD2. All rights reserved.
//

import Foundation
import Shared
import SwiftUI

extension QuestionChoice {

    func image() -> Image {
        switch self.game {
        case .design:
            return  R.image.rectangle_1.image
        case .product:
            return  R.image.rectangle_2.image
        default:
            return R.image.rectangle_3.image
        }
    }
}

extension QuestionChoice: Identifiable {

}

extension Array where Element == QuestionChoice {
    func nbOkResponse() -> Double {
        Double(self.filter {
            $0.userSelectedAnswer?.isCorrect == true
        }.count)
    }

    func nbKoResponse() -> Double {
        Double(self.filter {
            $0.userSelectedAnswer?.isCorrect == false
        }.count)
    }

    func averageMsResponseTime() -> Double {
        let sumOfTime = Double(self.reduce(0) { $0 + $1.responseTime })
        let totalTime = (sumOfTime / Double(self.count)).rounded()
        return totalTime < 1 ? 1 : totalTime
    }
}

File: androidApp/src/main/java/com/cacd2/cacdgame/android/model/QuestionChoice.kt
package com.cacd2.cacdgame.android.model

import androidx.annotation.DrawableRes
import com.cacd2.cacdgame.android.R
import com.cacd2.cacdgame.model.GameChoice
import com.cacd2.cacdgame.model.QuestionChoice

/**
 * Created by francois.dabonot@cacd2.fr on 14/06/2023.
 */

val QuestionChoice.image: Int
    @DrawableRes
    get() =
        when (this.game) {
            GameChoice.DESIGN -> R.drawable.rectangle_1
            GameChoice.PRODUCT -> R.drawable.rectangle_2
            GameChoice.TECH -> R.drawable.rectangle_3
            GameChoice.ALL -> R.drawable.rectangle_2
        }
